src/Toolbox.php

```diff
@@ -2109,83 +2111,94 @@ public static function createSchema($lang = 'en_GB', ?DBmysql $database = null)
         /** @var \DBmysql $DB */
         $DB = $database;
 
-        if (!$DB->runFile(sprintf('%s/install/mysql/glpi-empty.sql', GLPI_ROOT))) {
-            echo "Errors occurred inserting default database";
-        } else {
-           //dataset
-            Session::loadLanguage($lang, false); // Load default language locales to translate empty data
-            $tables = require_once(__DIR__ . '/../install/empty_data.php');
-            Session::loadLanguage('', false); // Load back session language
-
-            foreach ($tables as $table => $data) {
-                $reference = array_replace(
-                    $data[0],
-                    array_fill_keys(
-                        array_keys($data[0]),
-                        new QueryParam()
-                    )
-                );
+        $queries = $DB->getQueriesFromFile(sprintf('%s/install/mysql/glpi-empty.sql', GLPI_ROOT));
 
-                $stmt = $DB->prepare($DB->buildInsert($table, $reference));
+        foreach ($queries as $query) {
+            if ($progressCallback) {
+                $progressCallback();
+            }
+            if (!$query) {
+                continue;
+            }
```

Comment:
`getQueriesFromFile` should filter "empty" entries in order to not have to check here if the entry is valid.

```suggestion
```